### PR TITLE
Add slide regarding GitLab CI Pipeline Efficiency

### LINKED ISSUE #201: Adapt CI to switch to the python container at a later point
It cool to show that you can write bash in the container and install and use stuff to alpine, but it would make sense to later switch to a python image

### PR DESCRIPTION
**Pull request by martialblog**
Since a Lab would take up some time, I decided against it for now.

However, if time allows, we could add a lab in the future.

Fixes #201

### CODE REVIEW

File: day2/03_CI/03_Gitlab_CI.md [hunk 1]
```diff
@@ -627,6 +627,23 @@ Navigate into the repository > `CI / CD` > Jobs > `#...`  and choose `Job Artifa
...
 This is an example of how to do it from a CLI, the Gitlab WebIDE is an obvious alternative.
 
+!SLIDE
+# GitLab CI Pipeline Efficiency
+
+There are various ways to optimize GitLab CI pipelines.
+
+* Fail early to avoid wasting resources
+* Use caching for depencencies
+* Use Container Images with depencencies pre-installed
```
Comment (mocdaniel) [inline day2/03_CI/03_Gitlab_CI.md:637]:
```suggestion
* Use caching for dependencies
* Use Container Images with dependencies pre-installed
```

**Review by mocdaniel — CHANGES_REQUESTED** [review]
Found two small typos.

**Comment by martialblog** [comment]
Fixed. Thanks!

**Review by mocdaniel — APPROVED** [review]

**Comment by mocdaniel** [comment]
LGTM

**Pull request merged**